### ISSUE TITLE
Allow CrossRef search in BibEdit with FREE "Query Services account"

### ISSUE THREAD

**Issue by sbujam** [inveniosoftware/invenio#1277]
Originally on 2013-01-31

The newly introduced CrossRef utils allow registered CrossRef users to search -and import- records into Invenio by just entering the DOI. This was a highly anticipated addon!

For end users to be able to use it, Invenio admin must specify a CrossRef username/password in invenio(-local).conf Unfortunately, such an account is a privilege of CrossRef members only, a service that comes with a premium (let alone all the paperwork that needs to be filled).

Luckily, there is an alternative to becoming a full member and still be able to search CrossRef via OpenURL, using a FREE Query Services account! The only 'catch' is that this account allows up to a ~10^5 searches/year (more than enough for most of us...)

One must simply register (and verify) an email here: http://www.crossref.org/requestaccount/
that will be used as the 'username' for the searches. NO password is required!

The syntax for searching CrossRef with a free "Query Services account" is slightly different and described here: http://help.crossref.org/#using_the_open_url_resolver

So, Invenio could easily take advantage of this no-paperwork/free service from CrossRef and feed data into BibEdit with just the following trivial crossrefutils.py patch:

```
--- crossrefutils.py    2013-01-31 19:22:08.278990797 +0200
+++ crossrefutils.py.orig       2013-01-31 19:14:36.222417135 +0200
@@ -52,10 +52,8 @@
     doi = doi.strip()

     # Getting the data from external source
-    url = "http://www.crossref.org/openurl/?pid=" +  CFG_CROSSREF_USERNAME
-    if CFG_CROSSREF_PASSWORD:
-       url += ":" + CFG_CROSSREF_PASSWORD
-    url += "&noredirect=tru&id=doi:" + doi
+    url = "http://www.crossref.org/openurl/?pid=" +  CFG_CROSSREF_USERNAME \
+        + ":" + CFG_CROSSREF_PASSWORD + "&noredirect=tru&id=doi:" + doi
     request = urllib2.Request(url)
     response = urllib2.urlopen(request)
     header = response.info().getheader('Content-Type')
```

invenio.conf could also get an updated explanatory text like:

```
## CFG_CROSSREF_USERNAME -- the username used when sending request
## to the Crossref site.
## In case a Query Services account is used, enter the registered email instead.
CFG_CROSSREF_USERNAME =

## CFG_CROSSREF_PASSWORD -- the password used when sending request
## to the Crossref site.
## In case a Query Services account is used, please leave this field empty.
CFG_CROSSREF_PASSWORD =
```

**Comment by chokribr** [comment]
@sbujam i try to use it but always i've got this error This is not a correct DOI, please correct it 
note that i'm using invenio 1.2

**Comment by tiborsimko** [comment]
CC @kaplun see also https://github.com/inveniosoftware/invenio/pull/3516/files

**Comment by kaplun** [comment]
Oh! This is exactly @chokribr solution.
I have now sent a PR with a single patch co-authored by me, @sbujam and @chokribr that implements both free account support and proper encoding.

**Comment by tiborsimko** [comment]
Closed by #3516.
